src/containers/quiz/quiz-question/TutorAnswerGrading.tsx

```diff
@@ -0,0 +1,156 @@
+import { Box, IconButton, Stack } from '@mui/material'
+import { useTranslation } from 'react-i18next'
+import CheckIcon from '@mui/icons-material/Check'
+import CloseIcon from '@mui/icons-material/Close'
+import { styles } from '~/containers/quiz/quiz-question/Question.styles'
+import { useCallback, useEffect, useState } from 'react'
+import { ResourceService } from '~/services/resource-service'
+import { useParams } from 'react-router-dom'
+import useQuery from '~/hooks/use-query'
+import { ONE_HOUR, snackbarVariants } from '~/constants'
+import useMutation from '~/hooks/use-mutation'
+import { type ErrorResponse } from '~/types'
+import { useAppDispatch } from '~/hooks/use-redux'
+import { openAlert } from '~/redux/features/snackbarSlice'
+import { getErrorKey } from '~/utils/get-error-key'
+
+interface TutorAnswerGradingProps {
+  questionText?: string
+  onUpdate?: (isCorrect: boolean) => void
+}
+
+const TutorAnswerGrading: React.FC<TutorAnswerGradingProps> = ({
+  questionText,
+  onUpdate
+}) => {
+  const { t } = useTranslation()
+  const { id: cooperationId = '', quizId = '', attemptId = '' } = useParams()
+  const [isCorrect, setIsCorrect] = useState<boolean>(false)
+  const dispatch = useAppDispatch()
+
+  const getFinishedQuizzes = useCallback(() => {
+    return ResourceService.getFinishedQuizzesByQuizId(cooperationId, quizId)
+  }, [cooperationId, quizId])
+
+  const { data: finishedQuizzes = [], isLoading } = useQuery({
+    queryKey: ['finished-quizzes', cooperationId, quizId],
+    queryFn: getFinishedQuizzes,
+    options: {
+      staleTime: ONE_HOUR
+    }
+  })
+
+  useEffect(() => {
+    if (!isLoading && finishedQuizzes.length > 0 && questionText) {
+      const currentQuiz = finishedQuizzes.find((quiz) => quiz._id === attemptId)
+      if (currentQuiz) {
+        const questionResult = currentQuiz.results.find(
+          (result) => result.question === questionText
+        )
+        if (questionResult && questionResult.answers.length > 0) {
```

Comment:
Again, inverse condition
```suggestion
        if (!questionResult) {
          return 
```